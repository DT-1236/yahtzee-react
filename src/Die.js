import React, { Component } from 'react';
import styled from 'styled-components';

const Button = styled.button`
  font-size: 25px;
  display: inline-block;
  padding: 0.5em 0.8em;
  font-family: monospace;
  font-weight: bold;
  background-color: red;
  color: white;
  margin-right: 1em;

  &.Die-locked {
    background-color: darkred;
  }
`;

class Die extends Component {
  render() {
    return (
      <Button
        className={this.props.locked ? 'Die Die-locked' : 'Die'}
        onClick={() => this.props.handleClick(this.props.idx)}
      >
        {this.props.val}
      </Button>
    );
  }
}

export default Die;
