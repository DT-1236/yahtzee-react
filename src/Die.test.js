import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Die from './Die';

describe('simple smoke tests', () => {
  it('renders without crashing', () => {
    shallow(<Die />);
  });

  it('mounts without crashing', () => {
    mount(<Die />);
  });
});

describe('simple snapshots', () => {
  it('matches the shallow snapshot', () => {
    const wrapper = shallow(<Die />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});

describe('locked and unlocked snapshots', () => {
  it('matches the locked snapshot', () => {
    const wrapper = shallow(<Die locked />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('matches the unlocked snapshot', () => {
    const wrapper = shallow(<Die locked={false} />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
