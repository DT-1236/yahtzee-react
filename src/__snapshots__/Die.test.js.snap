// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`locked and unlocked snapshots matches the locked snapshot 1`] = `
<styled.button
  className="Die Die-locked"
  onClick={[Function]}
/>
`;

exports[`locked and unlocked snapshots matches the unlocked snapshot 1`] = `
<styled.button
  className="Die"
  onClick={[Function]}
/>
`;

exports[`simple snapshots matches the shallow snapshot 1`] = `
<styled.button
  className="Die"
  onClick={[Function]}
/>
`;
